File: messaging-service/internal/config/redis.go
package config

import (
	"context"
	"strings"

	"github.com/redis/go-redis/v9"
	"go.uber.org/zap"
)

func ConnectRedis(cfg *Config, logger *zap.Logger) (*redis.Client, error) {
	// Parse Redis URL
	redisURL := cfg.RedisURL
	redisURL = strings.TrimPrefix(redisURL, "redis://")

	client := redis.NewClient(&redis.Options{
		Addr:         redisURL,
		Password:     "", // no password set
		DB:           0,  // use default DB
		PoolSize:     50,
		MinIdleConns: 10,
	})

	// Test connection
	ctx := context.Background()
	if err := client.Ping(ctx).Err(); err != nil {
		return nil, err
	}

	logger.Info("Redis connected successfully")
	return client, nil
}

File: social-service/src/config/redis.ts
import { createClient, RedisClientType } from 'redis';
import logger from '../utils/logger';

let redisClient: RedisClientType | null = null;

export const connectRedis = async (): Promise<RedisClientType | null> => {
  try {
    if (!process.env.REDIS_URL) {
      logger.warn('Redis URL not configured, caching will be disabled');
      return null;
    }

    redisClient = createClient({
      url: process.env.REDIS_URL,
    }) as RedisClientType;

    redisClient.on('error', (err) => {
      logger.error('Redis Client Error:', err);
    });

    redisClient.on('connect', () => {
      logger.info('Redis Client Connected');
    });

    await redisClient.connect();
    return redisClient;
  } catch (error) {
    logger.error('Failed to connect to Redis:', error);
    return null;
  }
};

export const getRedisClient = (): RedisClientType | null => redisClient;

